snakemake/io.py

```diff
@@ -733,10 +733,10 @@ def apply_wildcards(self, wildcards):
                     validation_res,
                     rule=self.rule,
                 )
-
-            file_with_wildcards_applied = IOFile(
-                str(storage_object.local_path()), rule=self.rule
-            )
+            local_path_str = str(storage_object.local_path())
+            if ON_WINDOWS:
+                local_path_str = local_path_str.replace("\\", "/")
+            file_with_wildcards_applied = IOFile(local_path_str, rule=self.rule)
```

Comment:
The Path object has a method `as_posix()` which does exactly what you do here. I don't understand though why we need this at all. What is the problem with having a windows path returned here?